### DISCUSSION TITLE
Cactus comments not loading assets (css and js)

### DISCUSSION THREAD

**Discussion by krinkko** [CaiJimmy/hugo-theme-stack#1090] [General]
As cactus is currently not loading via their CDN, would it be possible to include the css and js from their gitlab into the project?

**Comment by krinkko** [comment]
I think the repo is here:

https://gitlab.com/cactus-comments/cactus-client

**Reply by ElastiCourse** [reply]
I haven't personally used cactus, but I do see 1 reference for the js in `data/external.yaml`. Since their CDN doesn't load the file, you may be able to workaround this issue by downloading the file (https://gitlab.com/cactus-comments/cactus-client/-/raw/main/src/cactus.js?ref_type=heads&inline=false) and placing it in `static` folder. Update `data/external.yaml` by replacing CDN URL `src: https://latest.cactus.chat/cactus.js` with `src: /cactus.js`.

**Reply by krinkko** [reply]
Thank you! Now it's loading the files but I get a reference error after updating the site:

`Uncaught ReferenceError: initComments is not defined`

**Reply by ElastiCourse** [reply]
> initComments is not defined

I can replicate this error on my end as well. Something feels off about their service at this point, including their own demo page https://cactus.chat/demo/.

The plugin has not been updated in 3 years. You may have better luck with a more stable and maintained comments system like Giscus.
